crates/cli/tests/dynamic_linking_test.rs

```diff
@@ -39,6 +39,22 @@ pub fn test_dynamic_linking() -> Result<()> {
     Ok(())
 }
 
+#[test]
+fn test_producers_section_present() -> Result<()> {
+    let js_wasm = create_dynamically_linked_wasm_module("console.log(42)")?;
+    let producers_section = wasmparser::Parser::new(0)
+        .parse_all(&js_wasm)
+        .find_map(|payload| match payload {
+            Ok(wasmparser::Payload::CustomSection(c)) if c.name() == "producers" => Some(c.data()),
+            _ => None,
+        })
+        .unwrap();
+    let producers_string = str::from_utf8(producers_section)?;
```

Comment:
The data section contents isn't a UTF-8 string but contains a series of UTF-8 strings. But the non-UTF8 parts are just unsigned LEB128 values that are `1` or `2` so this works as a very easy way to parse the byte array so we can do a quick inspection at test time.